### PR TITLE
MAINT/BUG: Don't squash useful error messages in favor of generic ones

### PR DESCRIPTION
**Pull request by eric-wieser**
Also avoid possible segfaults. Raised by discussion in #9924, but does not fix that.

### CODE REVIEW

File: numpy/core/src/multiarray/arraytypes.c.src [hunk 1]
```diff
@@ -356,18 +356,18 @@ string_to_long_double(PyObject*op)
...
             PyErr_Format(PyExc_ValueError,
-                         "invalid literal for long double: %s",
-                         s);
+                         "invalid literal for long double: %s (%s)",
+                         s,
+                         strerror(errno));
             Py_XDECREF(b);
             return 0;
         }
-        else if (errno) {
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/arraytypes.c.src]: These two `if`s are just swapped

**Comment by eric-wieser** [comment]
That doesn't look like a real failure

**Comment by eric-wieser** [comment]
[Problems with travis](https://www.traviscistatus.com/), apparently:

> Due to an upstream issue with our infrastructure provider, new builds on travis-ci.org are not starting.

**Comment by charris** [comment]
Is this going to be affected by @ahaldane's work?

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**